src/ucp/api/ucp_def.h

```diff
@@ -619,13 +619,15 @@ typedef ucs_status_t (*ucp_am_callback_t)(void *arg, void *data, size_t length,
  * @note If the incoming rendezvous data is not needed, the user can return
  *       UCS_OK from this routine without calling @ref ucp_am_data_recv_nbx.
  *       In this case data descriptor @a data will be dropped and the
- *       corresponding @ref ucp_am_send_nbx call should complete with
- *       UCS_ERR_REJECTED status. Otherwise, if UCS_INPROGRESS is returned from
- *       the routine delivering rendezvous data, the user takes ownership of
- *       data descriptor @a data. Then the user is responsible to either calling
- *       @ref ucp_am_data_recv_nbx for initiating data receive or
- *       @ref ucp_am_data_release for releasing the @a data descriptor.
- *
+ *       corresponding @ref ucp_am_send_nbx call should complete with the status
+ *       returned from the callback. Otherwise, if UCS_INPROGRESS is returned
+ *       from the routine delivering rendezvous data, the user takes ownership
+ *       of data descriptor @a data. Then the user is responsible to either
+ *       calling @ref ucp_am_data_recv_nbx for initiating data receive or
+ *       @ref ucp_am_data_release for releasing the @a data descriptor. If
+ *       @a data descriptor is released without invoking
+ *       @ref ucp_am_data_recv_nbx, the corresponding @ref ucp_am_send_nbx
+ *       operation will complete with UCS_ERR_CANCELED status.
```

Comment:
try to replace note by documenting return value. add UCS_ERR_xx to the list of possible return values.
return value specifies the behavior. in some cases calling recv_data+some return value is considered invalid usage